Asynchronous communication

miwashun opened the pull request:
非同期通信を導入しました
プルリクエストよろしくお願いします
https://i.gyazo.com/ca8e1a35ebac53a2e19dfb7ce619a217.mp4

Commit:
rspec

画像読み込めなかった

Commit:
first commit

Commit:
update

Commit:
modify

画像のみで投稿できるようにバリデーションの変更
それとテスト

Commit:
create

非同期通信の導入

Commit:
modify

Newアクションではなくcreateアクションに変更

yusukeyamane commented on app/controllers/messages_controller.rb:
```diff
@@ -9,7 +9,10 @@ def index
     if @message.save
-      redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'
+      respond_to do |format|
+        format.html { redirect_to group_messages_path(@group) , notice: 'メッセージが送信されました'}
```
動画を見た所非同期の場合はフラッシュメッセージが出ないという実装になっているので、手元では同期的な処理になっている気がします。

この行を消しても問題なく動作するかご確認頂けますか？

miwashun replied:
非同期通信できていなかったので非同期通信を設定しました
https://gyazo.com/5b64d2c9b191cd35ef0e9bcfd5766140

ReoKuriyama replied:
html形式でレスポンスを返さないので消してしまいましょう

miwashun replied:
削除しました

Commit:
Configuration

JavaScriptの設定

hzak2xx commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,43 @@
+                  <p class="message__text">
+                  </p>
+                  <p class="lower-message__content">
+                   ${message.content}
```
画像の表示がないと思うので追加しましょう

miwashun replied:
画像を非同期通信できました

hzak2xx commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,43 @@
+                    </p>
+                  </div>
+                  <p class="message__text">
+                  </p>
```
子要素がありませんがこのタグは必要なんですか？

miwashun replied:
とりあえず消しました

miwashun replied:
この行を消すと書式が崩れました

miwashun replied:
pタグからdivタグに変えcontentとimgタグをネストしました

hzak2xx commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,43 @@
+$(function(){
+  function buildHTML(message){
+    var html = `<div class="message" data-message_id>
```
data属性の指定ができていないかと思います

miwashun replied:
data属性を調べた結果今は必要なさそうなので削除しました

Commit:
delete

初回起動時に非同期にならないので
$(document).on('turbolinks:load', function() { });
を書き込んだが出来ず、メンターさんにやってもらっても改善されなかったのでとりあえずturbolinksを切りました

Commit:
Ajax

取り敢えずは非同期通信ができました

ReoKuriyama commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,48 @@
+      $('.form__message').val('')
+      $('.messages').animate({scrollTop: $('.messages')[0].scrollHeight}, 500, 'swing');
+      $('.form__submit').prop('disabled', false);
+
```
空行消しましょう

miwashun replied:
消しました

ReoKuriyama reviewed (changes requested):
rspecのファイルが混ざっていますね 👀 
このPRに混ざらないようにしましょう
ただし、今は技術的に難しいと思うので次回から気をつけましょう

hzak2xx commented:
プッシュしてください

Commit:
modify

同期通信を削除しました

Commit:
remove

スペースを取り除きました

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,46 @@
+      var html = buildHTML(data);
+      $('.messages').append(html)
+      $("#create_message")[0].reset();
+      $('.form__message').val('')
```
reset使いましょう

miwashun replied:
$('.form__message'.val('')を消して$("#create_message")[0].reset();だけにしました

Commit:
Remove

TOMARUMARU reviewed (approved):
LGTM
